[cutlass backend] check against arch >= 100

henrylhtsang opened the pull request:
Summary:
Want to add a guard against silent fallback to SM90.

GenerateSM100 was just added 3 days ago. https://github.com/NVIDIA/cutlass/blame/main/python/cutlass_library/generator.py#L8896

It should show up in CUTLASS 3.8 (not pinned yet).

Test Plan: ci

Differential Revision: D68748705




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

ColinPeppler commented on torch/_inductor/codegen/cuda/cutlass_utils.py:
```diff
@@ -81,6 +81,13 @@ def try_import_cutlass() -> bool:
+            f"Unexpected cuda arch: {arch}, please report this issue."
+        )
+
+    if int(arch) >= 100:
+        return "100"
```
nit: follow the `elif` convention

Aidyn-A commented on torch/_inductor/codegen/cuda/cutlass_utils.py:
```diff
@@ -81,6 +81,13 @@ def try_import_cutlass() -> bool:
 
 
 def _normalize_cuda_arch(arch: str) -> str:
+    if int(arch) >= 110:
```
There is `sm120` that is supposed to work.

Aidyn-A replied:
I agree on the warning. But, I am not sure how to handle `cutlass_generator.GenerateSMXX`...

henrylhtsang replied:
no worries, thanks

henrylhtsang commented:
@Aidyn-A feel free to take another look

henrylhtsang committed:
[cutlass backend] check against arch >= 100 (#145812)

Summary:

Want to add a guard against silent fallback to SM90.

GenerateSM100 was just added 3 days ago. https://github.com/NVIDIA/cutlass/blame/main/python/cutlass_library/generator.py#L8896

It should show up in CUTLASS 3.8 (not pinned yet).

Test Plan: ci

Reviewed By: chenyang78, ColinPeppler

Differential Revision: D68748705

henrylhtsang commented:
@Aidyn-A?

Aidyn-A reviewed (approved):
I apologize gorgeous the delay. Now it looks like it will not break sm120, but just warn.

henrylhtsang commented:
@pytorchbot merge -i
